regcore/models.py

```diff
@@ -5,6 +5,15 @@ class PartQuerySet(models.QuerySet):
     def effective(self, date):
         return self.filter(date__lte=date).order_by("name", "-date").distinct("name")
 
+    def effective_title(self, date, title):
+        return self.filter(title=title).filter(date__lte=date).order_by("name", "-date").distinct("name")
+
+    def full_part(self, date, title, part):
+        return self.filter(title=title).filter(date__lte=date).filter(name=part).latest("date")
```

Comment:
These can be brought up for clarity: `Part.objects.effective(date).filter(title=title, part=part)`